tests/test_public_udfs.py

```diff
@@ -0,0 +1,92 @@
+import fused
+import math
+import mercantile
+from typing import Any
+import os
+import requests
+import pytest
+
+
+UDFS_THAT_ERROR = [
+    # server_rt2.endpoints.secret_manager.SecretKeyNotFound: ARRAYLAKE_TOKEN
+    "Arraylake_Example",
+    # Serialization error TODO: delete lines after deployment (already fixed in staging)
+    "DC_AOI_Example",
+    "Sentinel1_Tile_Example",
+    "SAR_Umbra_File_Example",
+    # Misconfigured UDFs not present in repo
+    "GOES_18_Runner",
+    "GOES_18_Async",
+    "GOES_18_Partitions",
+]
+
+
+def get_public_udf_folders():
+    """
+    Get the list of public UDF folders from the UDFs repo as pytest params.
+    Fetches the list based on the env variable "GITHUB_SHA".
+    """
+
+    headers = {}
+    params = {}
+    if os.environ.get("GITHUB_TOKEN"):
+        headers["Authorization"] = f"Bearer {os.environ.get('GITHUB_TOKEN')}"
+    if os.environ.get("GITHUB_SHA"):
+        params["ref"] = os.environ.get("GITHUB_SHA")
+        print(f"fetching UDFs from commit: {params['ref']}")
+
+    response = requests.get(
+        "https://api.github.com/repos/fusedio/udfs/contents/public",
+        headers=headers,
+        params=params,
+    )
+
+    if response.status_code != 200:
+        raise Exception(f"Failed to fetch UDFs: {response.status_code}")
+    contents = response.json()
+
+    NON_UDF_FOLDERS = ["community", "common"]
```

Comment:
```suggestion
    NON_PUBLIC_UDF = ["community", "common"]
```

Small nitpick, but those are also UDFs, just not the public ones we want to test 

Any reason why you're skipping `common`?

Reply:
Just realized that's also an UDF (I thought it was just for utils). Will update 👍
Also, `community` being in the `public` folder seems like a mistake.

Reply:
It's not though?
![CleanShot 2025-03-05 at 09 26 02@2x](https://github.com/user-attachments/assets/ed021136-55c5-4729-99c4-854eba1094b4)

Reply:
@MaxLenormand try looking into the community folder under the public folder - that's the one I feel is a mistake

Reply:
This one indeed seems wrong: https://github.com/fusedio/udfs/tree/main/public/community/pgzmnk/Fire_Proximity_Buffer

Reply:
Caught up with Surya: we're going to skip this one until we figure out if we can remove it / merge with the same one under `udfs/community/` with @pgzmnk

Reply:
@MaxLenormand, I had a chat with him. Have made the corresponding changes to delete it at #690